packages/extension/src/newtab/LinksForm.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { ReactElement, useEffect, useState } from 'react';
+import { TextField } from '@dailydotdev/shared/src/components/fields/TextField';
+
+const limit = 8;
+const list = Array(limit).fill(0);
+
+interface LinksFormProps {
+  links: string[];
+  isFormReadonly?: boolean;
+  errors?: Record<string | number, string>;
+}
+
+export function LinksForm({
+  links,
+  errors,
+  isFormReadonly,
+}: LinksFormProps): ReactElement {
+  const [staleInput, setStaleInput] = useState({});
+
+  useEffect(() => {
+    const unchanged = {};
+    Object.keys(errors).forEach((key) => {
+      unchanged[key] = true;
+    });
+    setStaleInput(unchanged);
+  }, [errors]);
+
+  const onChange = (i: number) => {
+    if (!staleInput[i]) {
+      return;
+    }
+
+    setStaleInput((state) => ({ ...state, [i]: false }));
+  };
+
+  return (
+    <div className="flex flex-col gap-4">
+      {list.map((_, i) => (
+        <TextField
```

Comment:
Why don't we apply for any local check as well?
Are we relying only on server-side validation?

EDIT: Giving it another look, I see we have local validations but we can use the built-in input type for doing this validation. I think it will be better than a js based solution

Reply:
Was thinking of not fully managing every field (callback and checks) with a separate state of their own since all of their validations will run through the same check and hints so I decided to create an object instead.

I tried checking the `ProfileForm` and it handles each input's validity separately, should we do the same for this scenario?